app/services/full_chemical_populator_service.rb

```diff
@@ -0,0 +1,77 @@
+class FullChemicalPopulatorService
+  CSV_FILE = Rails.root.join('db/full_chemicals.csv')
+
+  def call
+    FullChemical.unrestrict_primary_key
+
+    full_chemicals.each_slice(5000) do |values|
```

Comment:
No idea how much memory balloons (from memory this 65k csv rows?? is that right) up to with this but you could read CSVs iteratively (ie `CSV.foreach`) - build those into an array and flush (INSERT) to db every 5k records the once after the iteration finishes?

Reply:
It significant enough to justify this change I think